### PR TITLE
Reworked with new template, TS support, dependencies updated

### PR DESCRIPTION
**Pull request by m4rvr**
- Use current versions of storyblok modules (especially storyblok-js-client). 
- TypeScript now works correctly, extended `Vue` interface for use of `$storyapi` & `$storybridge`
- _'storyblok-nuxt' needs to be added to the `types` in `tsconfig.json` of the nuxt application for autocompletion_

### CODE REVIEW

**Comment by MarvinRudolph** [comment]
@onefriendaday Please have a look into it. Tested it with the current nuxt version and it worked fine. :) Thank you!

**Comment by onefriendaday** [comment]
Currently Nuxt still uses Corejs 2.x. To avoid conflicts I would use storybloks-js-client V1 as V2 is installing Corejs 3.x.

**Comment by MarvinRudolph** [comment]
> Currently Nuxt still uses Corejs 2.x. To avoid conflicts I would use storybloks-js-client V1 as V2 is installing Corejs 3.x.

Whoops I will try to fix it soon. :) Thank you!

**Comment by MarvinRudolph** [comment]
@onefriendaday Downgraded `storyblok-js-client` to `1.0.34` and added a separate branch with `2.0.2` named `core-js@3` for future. :)

**Comment by MarvinRudolph** [comment]
@onefriendaday Did some more work into it and updated the structure, TypeScript support, README and package.json. Would be nice if you can have a look into it. :)

**Comment by Gomah** [comment]
@onefriendaday It would be great to actually publish a `storyblok-nuxt@next ` version in prior to use core-js v3, what do you think?

**Comment by f3ltron** [comment]
As i am using a lot and plan to use a lot storyblok later i would like to help for the plugin. Can you publish a @next branch and some roadmap for us for contributing ?

**Comment by MarvinRudolph** [comment]
Kinda inactive :/

**Comment by DominikAngerer** [comment]
Will check with Alex. Currently many new core features are getting reviewed by him. 👍

**Comment by onefriendaday** [comment]
The PR has merge conflicts, please sync with master to resolve the conflicts.

**Comment by tillmon** [comment]
@MarvinRudolph any chance to resolve these conflicts? TS support would be great!

**Comment by DominikAngerer** [comment]
Will be reviewed this weekend :)

**Comment by emanuelgsouza** [comment]
Hello @MarvinRudolph! How are you? Until the moment, I have some changes requests before merge this PR:

- When you resolve the conflicts:
  - In the `package.json`, you can maintain the `storyblok-js-client` in v2, because the `corejs3` is required.
  - In the `package.json`, could you change te `version` field to `'0-0-development'`.
- Could you add a section in the README, explaining how to setup the `types` for autocomplete?
- Could you update the dependencies of the project?
- We use the [semantic-release](https://semantic-release.gitbook.io/semantic-release/) for releasing new versions. So, about these changes, could you add a commit of `feat` type?

Any doubts I'm available. Thanks!

**Comment by Lindar90** [comment]
Hi folks! Is there any way how I can help to finish this PR? 
CC: @emanuelgsouza @MarvinRudolph

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Module } from '@nuxt/types'
+import Storyblok from 'storyblok-js-client'
+import initOptions, { Options } from './options'
+import setupBuild from './build'
+
+declare module 'vue/types/vue' {
+  interface Vue {
+    $storyapi: Storyblok
+    $storybridge: StoryblokBridge
```
Comment (Lindar90) [inline src/index.ts:9]: I might be mistaken but it looks like there is no neither StoryblokBridge interface declaration nor StoryblokBridge interface import in this file

**Review by Lindar90 — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (MarvinRudolph) [reply src/index.ts:9]: Yeah, you're right. Don't know what happened there. 🤔

**Comment by MarvinRudolph** [comment]
@Lindar90 I think it might be better to drop the pull request and refactor it. It's nearly a year old and a lot has changed since then (in Nuxt at least AFAIK). And I haven't had much time to check that, sorry! What do you think? 
This doesn't mean I don't want to work on it, would be really cool to bring types for the modules out! It's on my list but not planned in the next weeks.

File: src/index.ts (hunk 1 above)
Comment (derfl007) [reply src/index.ts:9]: I'm a little late here, but it's still relevant, I feel like it's actually correct like that. `StoryblokBridge `is under `declare global` in the type declaration file of `storyblok-js-client`, so I think it's available without an explicit import, once `Storyblok `is imported in the file.

**Comment by alvarosabu** [comment]
Hi, @m4rvr sorry that this Draft wasn't updated in a while. The good news, we are working on the Typescript support atm  here https://github.com/storyblok/storyblok-nuxt/pull/296 . I will close this PR

**Pull request closed without merging**